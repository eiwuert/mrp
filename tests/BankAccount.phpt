<?php

use Tester\Assert;

require __DIR__ . '/bootstrap.php';

$mrp = new \Mrp\BankAccount();
$mrp->addRow([
	'idr' => '1',
    'idradr' => '1',
    'banka' => 'AAAAAAAAAAAAAAAAAAA',
    'pobocka' => 'AAAAAAAAAAAAAAAAAAA',
    'ucet' => 'AAAAAAAAAAAAAAAAAAA',
    'kodbanky' => 'AAAAAAAAAAAAAAAAAAA',
    'charkod' => 'AAAAAAAAAAAAAAAAAAA',
    'mena' => 'AAAAAAAAAAAAAAAAAAA',
    'specisymb' => 'AAAAAAAAAAAAAAAAAAA',
    'swiftcode' => 'AAAAAAAAAAAAAAAAAAA',
    'adresa2' => 'AAAAAAAAAAAAAAAAAAA',
	'adresa3' => 'AAAAAAAAAAAAAAAAAAA',
	'adresa4' => 'AAAAAAAAAAAAAAAAAAA',
    'iban' => 'AAAAAAAAAAAAAAAAAAA',
]);

Assert::exception(function () use ($mrp) {
	$mrp->addRow(['idr' => '1']);
}, 'Mrp\BankAccountException', "Missing field 'idradr'");

Assert::exception(function () use ($mrp) {
	$mrp->addRow(['idr' => '1', 'idradr' => '2', 'unknown' => 'unknown']);
}, 'Mrp\BankAccountException', "Unknown field 'unknown'");

Assert::matchFile(__DIR__ . '/BankAccount.getXml().expected', $mrp->getXml());
