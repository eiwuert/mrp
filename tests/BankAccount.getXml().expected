<?xml version="1.0" encoding="windows-1250"?>
<document>
  <datasets>
    <dataset0>
      <rows>
        <row>
          <fields>
            <idr>1</idr>
            <idradr>1</idradr>
            <banka>AAAAAAAAAAAAAAAAAAA</banka>
            <pobocka>AAAAAAAAAAAAAAAAAAA</pobocka>
            <ucet>AAAAAAAAAAAAAAAAAA</ucet>
            <kodbanky>AAAAAAAAAAAA</kodbanky>
            <charkod>AAAAAAA</charkod>
            <mena>AAA</mena>
            <specisymb>AAAAAAAAAA</specisymb>
            <swiftcode>AAAAAAAAAAAA</swiftcode>
            <adresa2>AAAAAAAAAAAAAAAAAAA</adresa2>
            <adresa3>AAAAAAAAAAAAAAAAAAA</adresa3>
            <adresa4>AAAAAAAAAAAAAAAAAAA</adresa4>
            <iban>AAAAAAAAAAAAAAAAAAA</iban>
          </fields>
        </row>
      </rows>
    </dataset0>
  </datasets>
</document>
